### PR TITLE
Added KMS Driver developer guide

### PR DESCRIPTION
**Pull request by soren-da**
Added the developer guide for KMS drivers with the API specification pulled in from the Canton repo and examples for a KMS driver test suite based on our AWS KMS driver. Both for 2.9 and 2.10

### CODE REVIEW

File: docs/2.10.0/docs/canton/usermanual/kms/kms_driver_guide.rst [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+confidential flag being true, which means any sensitive information in the
+configuration such as credentials, should be omitted from the written
+configuration. A specific test case should be added if your driver-specific
+configuration contains any confidential information, which asserts that the
+sensitive information is omitted.
+
+Run Canton with a KMS Driver
+----------------------------
+
+Configure Canton to run with a KMS driver, for example, with for a
```
Comment (JoaoSa-da) [inline docs/2.10.0/docs/canton/usermanual/kms/kms_driver_guide.rst:253]: `with for` sounds strange, probably just `for`

**Review by JoaoSa-da — APPROVED** [review]

File: docs/2.10.0/docs/canton/usermanual/kms/kms_driver_guide.rst [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+
+Concretely the Scala interface is defined as the following:
+
+.. literalinclude:: /canton/includes/mirrored/enterprise/kms-driver-api/src/main/scala/com/digitalasset/canton/crypto/kms/driver/api/v1/KmsDriver.scala
+   :language: Scala
+
+Error Handling and Health
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In case the driver experiences an error the Future of the operation should be
```
Comment (JoaoSa-da) [inline docs/2.10.0/docs/canton/usermanual/kms/kms_driver_guide.rst:126]: should fail

File: docs/2.10.0/docs/canton/usermanual/kms/kms_driver_guide.rst [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+failed with a ``KmsDriverException``. When the exception's flag retryable is
+true the caller side, i.e., Canton, will perform a retry with exponential
```
Comment (JoaoSa-da) [inline docs/2.10.0/docs/canton/usermanual/kms/kms_driver_guide.rst:128]: flag retryable is true, (add a comma)

**Review by JoaoSa-da — APPROVED** [review]

File: docs/2.10.0/docs/canton/usermanual/kms/kms_driver_guide.rst [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+
+.. literalinclude:: /canton/includes/mirrored/enterprise/aws-kms-driver/src/test/scala/com/digitalasset/canton/nightly/AwsKmsDriverTest.scala
+   :language: Scala
+   :start-after: user-manual-entry-begin: AwsKmsDriverTest
+   :end-before: user-manual-entry-end: AwsKmsDriverTest
+
+Given that generation of new keys can be expensive when running tests during
+development frequently, in particular when using Cloud-based KMSs, the test
+suite can also be configured to use predefined keys to test most parts (except
+key generation obviously) of the KMS Driver API:
```
Comment (carrielaben-da) [inline docs/2.10.0/docs/canton/usermanual/kms/kms_driver_guide.rst:220]:
```suggestion
Generating new keys can be expensive when running tests during
development, in particular when using cloud-based KMSs. To mitigate this, the test
suite can also be configured to use predefined keys to test most parts (except
key generation) of the KMS Driver API:
```

File: docs/2.10.0/docs/canton/usermanual/kms/kms_driver_guide.rst [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+
+.. literalinclude:: /canton/includes/mirrored/enterprise/aws-kms-driver/src/test/scala/com/digitalasset/canton/nightly/AwsKmsDriverTest.scala
+   :language: Scala
+   :start-after: user-manual-entry-begin: AwsKmsDriverWithPredefinedKeysTest
+   :end-before: user-manual-entry-end: AwsKmsDriverWithPredefinedKeysTest
+
+
+For each supported signing/encryption key specification an existing key alias/id
+can be configured as part of the predefined keys maps. When running the test
+suite the generation of new keys is not allowed.
```
Comment (carrielaben-da) [inline docs/2.10.0/docs/canton/usermanual/kms/kms_driver_guide.rst:230]:
```suggestion
For each supported signing/encryption key specification an existing key alias/ID
can be configured as part of the predefined key maps. When running the test
suite the generation of new keys is not allowed.
```

File: docs/2.10.0/docs/canton/usermanual/kms/kms_driver_guide.rst [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
+
+KmsDriverFactoryTest
+^^^^^^^^^^^^^^^^^^^^
+
+The test suite for the KMS driver factory is structured similarly:
```
Comment (carrielaben-da) [inline docs/2.10.0/docs/canton/usermanual/kms/kms_driver_guide.rst:235]:
```suggestion
The test suite for the KMS driver factory is structured similarly to the above:
```

File: docs/2.10.0/docs/canton/usermanual/kms/kms_driver_guide.rst [hunk 7]
```diff
@@ -0,0 +1,270 @@
...
+   :start-after: user-manual-entry-begin: AwsKmsDriverFactoryTest
+   :end-before: user-manual-entry-end: AwsKmsDriverFactoryTest
+
+
+The ``KmsDriverFactory`` can write the driver-specific configuration with a
+confidential flag being true, which means any sensitive information in the
+configuration such as credentials, should be omitted from the written
+configuration. A specific test case should be added if your driver-specific
+configuration contains any confidential information, which asserts that the
+sensitive information is omitted.
```
Comment (carrielaben-da) [inline docs/2.10.0/docs/canton/usermanual/kms/kms_driver_guide.rst:248]:
```suggestion
The ``KmsDriverFactory`` can write the driver-specific configuration with a
confidential flag being true, which means any sensitive information in the
configuration such as credentials should be omitted from the written
configuration. A specific test case should be added if your driver-specific
configuration contains any confidential information, asserting that the
sensitive information is omitted.
```

File: docs/2.10.0/docs/canton/usermanual/kms/kms_driver_guide.rst [hunk 8]
```diff
@@ -0,0 +1,270 @@
...
+.. literalinclude:: /canton/includes/mirrored/enterprise/app/src/test/resources/aws-kms-driver.conf
+   :language: none
+   :start-after: user-manual-entry-begin: AwsKmsDriverConfig
+   :end-before: user-manual-entry-end: AwsKmsDriverConfig
+
+Run Canton with your driver jar on its class path:
+
+``java -cp driver.jar:canton.jar com.digitalasset.canton.CantonEnterpriseApp -c canton.conf # further canton arguments``
+
+Where canton.jar depends on the canton version, e.g.,
```
Comment (carrielaben-da) [inline docs/2.10.0/docs/canton/usermanual/kms/kms_driver_guide.rst:265]:
```suggestion
Where canton.jar depends on the Canton version, e.g.,
```

**Review by carrielaben-da — APPROVED** [review]
Same changes to the 2.9 file.

File: docs/2.10.0/docs/canton/usermanual/kms/kms_driver_guide.rst (hunk 5 above)
Comment (soren-da) [reply docs/2.10.0/docs/canton/usermanual/kms/kms_driver_guide.rst:229]: there can be multiple keys in each map, so I kept the plural and only upper-cased the ID.

File: docs/2.10.0/docs/canton/usermanual/kms/kms_driver_guide.rst (hunk 2 above)
Comment (soren-da) [reply docs/2.10.0/docs/canton/usermanual/kms/kms_driver_guide.rst:126]: actually the future should be completed as a failure with the exception, so I think `should be failed` is correct here.

**Pull request merged**